File: app/entry/src/main/ets/models/Hitokoto.ts
export interface HitokotoPostRequest {
  hitokoto: string;
  type: string;
  source: string;
  creatorId: number;
  sourceWho: string | undefined;
}

export interface HitokotoResponse {
  message: string;
  hitokoto: Hitokoto | undefined;
}

export interface HitokotoListResponse {
  message: string;
  hitokotoList: Array<Hitokoto>
}

export interface Hitokoto {
  id: number;
  uuid: string;
  hitokoto: string;
  type: string;
  source: string;
  creatorId: number;
  createAt: number;
  length: number;
  sourceWho: string | undefined;
}

File: server/lib/models/hitokoto/hitokoto.dart
import 'package:server/index.dart';

part 'hitokoto.freezed.dart';
part 'hitokoto.g.dart';

@freezed
class HitokotoPostRequest with _$HitokotoPostRequest {
  const factory HitokotoPostRequest({
    required String hitokoto,
    required String type,
    required String source,
    required int creatorId,
    String? sourceWho,
  }) = _HitokotoPostRequest;

  factory HitokotoPostRequest.fromJson(Json json) =>
      _$HitokotoPostRequestFromJson(json);

  static Future<HitokotoPostRequest?> fromRequest(Request request) async {
    try {
      final json = await request.json();
      return HitokotoPostRequest.fromJson(json as Json);
    } catch (e) {
      return null;
    }
  }
}

@freezed
class HitokotoResponse with _$HitokotoResponse {
  const factory HitokotoResponse({
    required String message,
    Hitokoto? hitokoto,
  }) = _HitokotoResponse;

  factory HitokotoResponse.fromJson(Map<String, dynamic> json) =>
      _$HitokotoResponseFromJson(json);
}
